recipes/hwloc/all/test_package/test_package.c

```diff
@@ -0,0 +1,17 @@
+#include <hwloc.h>
+#include <stdio.h>
+
+int main(void) {
+  hwloc_topology_t topology;
+  int nbcores;
+  
+  hwloc_topology_init(&topology);  // initialization
+  hwloc_topology_load(topology);   // actual detection
```

Comment:
I would suggest reducing the test to calling only `hwloc_topology_init` and `hwloc_topology_destroy`, as I can see those are functions that are inside the library.

Since this file is not checking for the int output of any of the functions, and it's explicitly returning 0 at the end of the control flow, it's superfluous to call any other functionality, since  we are not testing correct behaviour. This is good - where possible, we want the test_package to test the minimum necessary to make sure that we can compile and link. i believe that is achieved by the first function altogether and there's no need for more.